### PR TITLE
feat(compiler): expose extension uri parameter in constructor

### PR DESCRIPTION
**Pull request by rtpsw**

### CODE REVIEW

File: ibis_substrait/compiler/core.py [hunk 1]
```diff
@@ -21,10 +21,12 @@ def which_one_of(message: msg.Message, oneof_name: str) -> tuple[str, Any]:
 
 
 class SubstraitCompiler:
-    def __init__(self) -> None:
+    def __init__(self, uri: str = None) -> None:
```
Comment (cpcloud) [inline ibis_substrait/compiler/core.py:24]:
```suggestion
    def __init__(self, uri: str | None = None) -> None:
```

**Review by cpcloud — CHANGES_REQUESTED** [review]

**Comment by cpcloud** [comment]
@rtpsw Can you fixup and change the commit to match the PR title (I changed the PR title).

**Comment by rtpsw** [comment]
@cpcloud, is anything remaining after the recent commit?

**Comment by rtpsw** [comment]
@cpcloud: is this ready to be merged?

**Comment by cpcloud** [comment]
@rtpsw Can you squash to a single commit matching the PR title?

**Comment by rtpsw** [comment]
I opted to add a commit with this comment and with an addition doc. I believe GH should allow you to squash using this comment.

**Comment by rtpsw** [comment]
Last commit fixes a [lint error](https://github.com/ibis-project/ibis-substrait/runs/5738580590?check_suite_focus=true).

**Comment by rtpsw** [comment]
@cpcloud: Hopefully I have the commit as required now.

**Comment by rtpsw** [comment]
To keep things simple, I created a clean PR https://github.com/ibis-project/ibis-substrait/pull/137 to replace this PR.

**Comment by rtpsw** [comment]
Canceling this.

**Pull request closed without merging**